components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/IdentityClaimValueEncryptionListener.java

```diff
@@ -198,6 +211,28 @@ private String encryptClaimValue(String plainText) throws CryptoException {
         return CryptoUtil.getDefaultCryptoUtil().encryptAndBase64Encode(plainText.getBytes(StandardCharsets.UTF_8));
     }
 
+    /**
+     * Encrypt claim value with custom key.
+     *
+     * @param plainText text to be encrypted.
+     * @return encrypted claim value.
+     * @throws CryptoException  If an error occurs while encrypting the claim value.
+     */
+    private String encryptClaimValueWithCustomKey(String plainText) throws CryptoException {
+
+        if (plainText.isEmpty()) {
```

Comment:
If we update the kernel to use same encrypt method, we should be able to refactor this code more. Refer kernel PR for more details